SampleViewer/DownloadsManager.h

```diff
@@ -0,0 +1,154 @@
+// [Legal]
+// Copyright 2025 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef DOWNLOADSMANAGER_H
+#define DOWNLOADSMANAGER_H
+
+class DataItem;
+class OfflineDataProjectsModel;
+class Sample;
+class SampleListModel;
+class QTimer;
+
+namespace Esri::ArcGISRuntime
+{
+  class Portal;
+  class PortalItem;
+}
+
+#include <QMap>
+#include <QObject>
+#include <QQueue>
+#include <QString>
+
+Q_MOC_INCLUDE("OfflineDataProjectsModel.h")
+
+class DownloadsManager : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(bool downloadInProgress READ downloadInProgress NOTIFY downloadInProgressChanged)
+  Q_PROPERTY(bool isBulkDownload READ isBulkDownload NOTIFY isBulkDownloadChanged)
+  Q_PROPERTY(QString downloadText READ downloadText NOTIFY downloadTextChanged)
+  Q_PROPERTY(double downloadProgress READ downloadProgress NOTIFY downloadProgressChanged)
+  Q_PROPERTY(bool cancelDownload READ cancelDownload WRITE setCancelDownload NOTIFY cancelDownloadChanged)
+  Q_PROPERTY(bool downloadFailed READ downloadFailed NOTIFY downloadFailedChanged)
+  Q_PROPERTY(OfflineDataProjectsModel* offlineDataProjects READ offlineDataProjects NOTIFY offlineDataStateChanged)
+
+public:
+  explicit DownloadsManager(QObject* parent = nullptr);
+  ~DownloadsManager() override;
+
+  // Initialization
+  void initialize(Esri::ArcGISRuntime::Portal* portal, const QString& homePath);
+
+  // Download operations
+  void downloadAllDataItems();
+  void downloadDataItemsForSample(Sample* sample);
+  void cancelAllDownloads();
+
+  // Delete operations
+  bool deleteAllOfflineData();
+  bool deleteProjectOfflineData(const QString& sampleName);
+
+  // Query methods
+  bool hasOfflineData(const QString& sampleName);
+  OfflineDataProjectsModel* getOfflineDataProjects();
+  bool hasAnyDataToDownload();
+  bool hasAnyDataToDelete();
+
+  // State getters
+  bool downloadInProgress() const;
+  bool isBulkDownload() const;
+  QString downloadText() const;
+  double downloadProgress() const;
+  bool cancelDownload() const;
+  bool downloadFailed() const;
+  OfflineDataProjectsModel* offlineDataProjects() const;
+
+  // State setters
+  void setCancelDownload(bool cancel);
+
+  void setSamples(SampleListModel* samples)
+  {
+    m_samples = samples;
+  }
```

Comment:
albeit simple code, we should avoid header implementations unless absolutely necessary.